src/NexusMods.Paths/Utilities/PathHelpers.cs

```diff
@@ -386,15 +386,24 @@ public static string JoinParts(ReadOnlySpan<char> left, ReadOnlySpan<char> right
         DebugAssertIsSanitized(right, os);
 
         var spanLength = GetExactJoinedPartLength(left, right);
-        var buffer = spanLength > 512
-            ? GC.AllocateUninitializedArray<char>(spanLength)
-            : stackalloc char[spanLength];
-
-        var count = JoinParts(buffer, left, right, os);
-        if (count == 0) return string.Empty;
-
-        Debug.Assert(count == spanLength, $"Calculated span length '{spanLength}' doesn't match actual span length '{count}'");
-        return buffer.ToString();
+        unsafe
+        {
+            fixed (char* leftPtr = left)
+            fixed (char* rightPtr = right)
+            {
+                return string.Create(spanLength, (((IntPtr)leftPtr, left.Length), ((IntPtr)rightPtr, right.Length), os), (span, tuple) =>
```

Comment:
The `Debug.Assert` call references `spanLength` so there's still a delegate allocation. This is also an issue with the string-counterpart of `JoinParts`.

Reply:
Yup. I don't consider it a big deal though. In release builds, where the `DEBUG` preprocessor definition isn't enabled, this code will be eliminated.